primitives/src/communities.rs

```diff
@@ -19,9 +43,38 @@ pub struct Location {
 }
 
 #[derive(Encode, Decode, Clone, PartialEq, Eq, Default, RuntimeDebug)]
-pub struct CommunityPropertiesType {
-    pub name_utf8: Vec<u8>,
-    pub demurrage_per_block: Demurrage,
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub struct CommunityMetadata {
+    /// utf8 encoded name
+    pub name: PalletString,
+    /// utf8 encoded abbreviation of the name
+    pub symbol: PalletString,
+    /// multi-resolution resource for the community icon
+    pub icons: Vec<Favicon>,
+    /// optional color scheme or other customizable styles to shape app appearance
+    pub theme: Option<Theme>,
+    /// optional link to a community site
+    pub url: Option<PalletString>,
+}
+
+#[derive(Encode, Decode, Clone, PartialEq, Eq, Default, RuntimeDebug)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub struct Favicon {
+    src: IpfsCid,
+    /// e.g. "64x64", "128x128", ...
+    sizes: PalletString,
```

Comment:
why is sizes plural if this type defines a single image resource?

Reply:
typo, but it gets removed anyway now.